rdrf/rdrf/models/definition/models.py

```diff
@@ -1639,14 +1639,18 @@ def naming_info(self):
 
     def clean(self):
         defaults = ContextFormGroup.objects.filter(registry=self.registry,
-                                                   is_default=True)
-        defaults.exclude(pk=self.pk)
+                                                   is_default=True).exclude(pk=self.pk)
         num_defaults = defaults.count()
+
         if num_defaults > 0 and self.is_default:
             raise ValidationError("Only one Context Form Group can be the default")
         if num_defaults == 0 and not self.is_default:
             raise ValidationError("One Context Form Group must be chosen as the default")
 
+        if self.naming_scheme == "C" and self.naming_cde_to_use is None:
```

Comment:
Technically this is wrong because an empty string fornaming cde would pass validation - write a mini parser method  to return a  triple of the objects needed - none otherwise 


(form_model, section_mode, cde_model)